pkg/perf/iperf/iperf_task.go

```diff
@@ -222,6 +223,31 @@ func (t *IperfTest) runIperfTest(ctx context.Context, clientIP string, tcp bool)
 	if !tcp && len(report.End.Streams) > 0 {
 		iperfResult.DownloadSpeed = report.End.Streams[0].UDP.BitsPerSecond
 	}
+	var speedtestServer *speedtest.Server
```

Comment:
this will result for unnecessary speedtest runs, the `runIperfTest`  function runs for each public node 4 times (tcp/udp, v4/v6) i think this is not needed in this test. 

also overriding the iperf result could be misleading, "if someone for any reason searching for iperf results"

i recommend doing speedtest as a standalone monitor task